core/jvm/src/main/scala/zio/internal/PlatformSpecific.scala

```diff
@@ -105,6 +105,9 @@ private[zio] trait PlatformSpecific {
   final def newConcurrentWeakSet[A]()(implicit unsafe: zio.Unsafe): JSet[A] =
     Collections.synchronizedSet(newWeakSet[A]())
 
+  final def newConcurrentWeakHashSet[A](implicit unsafe: zio.Unsafe): ConcurrentWeakHashSet[A] =
```

Comment:
Rather than adding this new method, let's just replace `newConcurrentWeakSet`.

Reply:
I didn't want to modify the previous method, because the change of return type to `ConcurrentWeakHashSet/mutable.Set` would be breaking, and I wasn't sure if you're fine with that.